packages/node-core/src/indexer/project.service.spec.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {NodeConfig} from '../configure';
+import {BaseDsProcessorService} from './ds-processor.service';
+import {DynamicDsService} from './dynamic-ds.service';
+import {BaseProjectService} from './project.service';
+import {ISubqueryProject} from './types';
+
+class TestProjectService extends BaseProjectService<any, any> {
+  packageVersion = '1.0.0';
+
+  async getBlockTimestamp(height: number): Promise<Date> {
+    return Promise.resolve(new Date());
+  }
+
+  onProjectChange(project: any): void {
+    return;
+  }
+}
+
+describe('BaseProjectService', () => {
```

Comment:
Can you please add tests over project versions to ensure there is only datasources from one version at any one time